crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -49,8 +53,12 @@ pub(crate) async fn check_stateful<S: StateReadExt + 'static>(
     state: &S,
 ) -> anyhow::Result<()> {
     let signer_address = Address::from_verification_key(tx.verification_key());
+    let fee_asset_id = tx
+        .unsigned_transaction()
+        .fee_asset_id
+        .unwrap_or(*NATIVE_ASSET.get().expect("native asset must be set").id());
```

Comment:
Mentioned this above: this could be made nicer using `Lazy` instead of `OnceCell`.

But even if you stick with once_cell, I recommend using something like `asset::get_native_asset()` or `native_asset:get()` and just have the panic in there.

Reply:
updated!